include/seastar/core/array_map.hh

```diff
@@ -31,19 +31,19 @@ namespace seastar {
 
 // unordered_map implemented as a simple array
 
-template <typename Value, size_t Max>
+template <typename Value, std::size_t Max>
```

Comment:
i will revert this change. as i've been working on C++ modules support. and found that these names in global namespaces are exposed by `std.compat` not `std`, hence this change.

Reply:
I agree it's "bad" (non-modern) C++ style to use size_t et al. instead of std::size_t, but since we have almost 2 thousand (!) other uses in Seastar, I think the two valid options is to either change all of them, or not at all. I don't see any point of changing it only in this specific header file.

Reply:
yeah, thought we could do this piecemeal. but yeah, we can also do this in a single pass when we are moving to a C++23 world where we have the access to `std.compat` and `std` !

Reply:
It's okay to do it either piecemeal or in one go. There's no reason to insist on strict methodology for such a simple change.

Reply:
@avikivity, I agree, but in that case it makes sense to:
1. Declare (and agree, and document it in the coding convention) that we want to do this change and why. Note that this "problem" isn't limit it to size_t - we also have 355 occurances of uint32_t and zero of std::unit32_t, and so on.
2. Not include this change as a separate patch in an irrelevant series.